### PR TITLE
Add validation to the Confirm Password field.

### LINKED ISSUE #45: Signup/Verify that confirm-password matches password input value

### PR DESCRIPTION
**Pull request by UserOlena**
Added to validation to the `Confirm Password` field to verify if its value matches `Password` value.

At this stage, all the fields of the Sign Up page have validation feature:
1. Verify that the text field is not blank, as all the fields are required;
2. Verify that the input value passes the corresponding regex pattern based on the field type;
3. For the both Password and Confirm Password validate that both input values match;

If any of the validations fails, corresponding error message should be displayed.
Error message should disappear on Focus for associated text field.

Password validation should work in any order regardless of what password text field user filled first (example: 
   provide value to the Password text field first and  Confirm Password afterwards;
   provide value to the Confirm Password text field first and Password afterwards;
)

closes #45

### CODE REVIEW

File: client/src/pages/SignUp.js [hunk 1]
```diff
@@ -165,8 +197,7 @@ export function SignUp() {
                   error={userName.isEmpty || !userName.isValid}
                   helperText={
                     (userName.isEmpty && 'Username field is required') ||
-                    (!userName.isValid &&
-                      'Username should consist of 4 to 12 alphanumeric characters.')
+                    (!userName.isValid && notValidUserNameErrorMessage)
```
Comment (sarah-jensen) [inline client/src/pages/SignUp.js:200]: line 200: error message for invalid username is not showing

File: client/src/pages/SignUp.js [hunk 2]
```diff
@@ -189,7 +220,7 @@ export function SignUp() {
                   helperText={
                     (email.isEmpty && 'Email field is required') ||
                     (!email.isValid &&
```
Comment (sarah-jensen) [inline client/src/pages/SignUp.js:222]: line 222: error message for invalid email is not showing

**Review by sarah-jensen — COMMENTED** [review]
It wasn't part of this commit, but I noticed in the console that the default state you're logging on line 145 (in validation.js) for `confirmPassword.isEmpty` is 'false' when the field is empty upon initial page load

**Comment by UserOlena** [comment]
> 



> It wasn't part of this commit, but I noticed in the console that the default state you're logging on line 145 (in validation.js) for `confirmPassword.isEmpty` is 'false' when the field is empty upon initial page load

Yes, that is expected behavior. If default value for all the text fields will be `true` for `isEmpty` that will lead to the error messages to be displayed upon the page load for all the text fields

**Review by sarah-jensen — APPROVED** [review]

**Pull request merged**